python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
-    name: str
-    description: str
-    column_details: list
-    retention_period_in_days: int | None
-    domain: str | None = None
-    parent_entity_name: str | None = None
-    relationships: dict[RelationshipType, list[RelatedEntity]] | None = None
-    source_dataset_name: str = ""
-    where_to_access_dataset: str = ""
-    fully_qualified_name: str | None = None
-    data_sensitivity_level: SecurityClassification = SecurityClassification.OFFICIAL
-    tags: list[str] = field(default_factory=list)
-    major_version: int = 1
-    row_count: int | None = None
-    owner: str = ""
-    owner_email: str = ""
-    last_updated: datetime | None = None
-    first_created: datetime | None = None
-
-    @staticmethod
-    def from_data_product_schema_dict(
-        metadata: dict[str, Any], table_name, retention_period: int | None = None
-    ):
-        """
-        Expects a dict containing data product table schema information as per the
-        required fields in the json schema at
-        https://github.com/ministryofjustice/modernisation-platform-environments/tree/main/terraform/environments/data-platform/data-product-table-schema-json-schema
-
-        and should be passed table name and optionally a retention period
-
-        Then populates a TableMetadata object with the given data.
-        """
-
-        new_metadata = TableMetadata(
-            name=table_name,
-            description=metadata["tableDescription"],
-            column_details=metadata["columns"],
-            retention_period_in_days=retention_period,
-            source_dataset_name=metadata.get("sourceDatasetName", ""),
-            where_to_access_dataset=metadata.get("sourceDatasetLocation", ""),
-            data_sensitivity_level=SecurityClassification[
-                metadata.get("securityClassification", "OFFICIAL")
-            ],
-            tags=metadata.get("tags", []),
-        )
-
-        return new_metadata
-
-
-# jscpd:ignore-start
-@dataclass
-class DatabaseMetadata:
-    """
-    For source system databases - currently matches DataProductMetadata
-    but will need to be revisted and refined potentially
-    """
-
-    name: str
-    description: str
-    version: str
-    owner: str
-    owner_display_name: str
-    maintainer: str | None
-    maintainer_display_name: str | None
-    email: str
-    retention_period_in_days: int
-    domain: str
-    subdomain: str | None
-    dpia_required: bool
-    dpia_location: str | None
-    last_updated: datetime
-    creation_date: datetime
-    s3_location: str | None
-    status: DataProductStatus = DataProductStatus.DRAFT
-    tags: list[str] = field(default_factory=list)
-
-
-# jscpd:ignore-end
-
-
-@dataclass
-class ChartMetadata:
-    name: str
-    description: str
-    external_url: str
-    tags: list[str] = field(default_factory=list)
+class Column(BaseModel):
+    """
+    A column definition in a table
+    """
+
+    name: str = Field(
+        pattern=r"^[\w.]+$",
```

Comment:
The dot matches any character, so should presumably be removed.
```suggestion
        pattern=r"^[\w]+$",
```